Ensure tags must fit pattern before adding them

mikemaccana opened the pull request:
Fixes #20

Commit:
Use .includes() rather than tilde & indexOf() trick
Space out a litle for clarity

Commit:
Don't add tag if it's invalid (eg, for pattern=). Fixes #20

Commit:
Build files and bump version

thomasdegry commented:
@developit any reason why we can't merge this yet? Would be awesome to have this functionality working!

developit commented:
`Array.prototype.includes()` is not supported in IE9, which this library targets.

developit commented on src/tags-input.js:
```diff
@@ -45,15 +45,26 @@ export default function tagsInput(input) {
 		// Add multiple tags if the user pastes in data with SEPERATOR already in it
-		if (~text.indexOf(SEPERATOR)) text = text.split(SEPERATOR);
-		if (Array.isArray(text)) return text.forEach(addTag);
+		if ( text.includes(SEPERATOR) ) {
```
seems a shame to give up IE compat here

mikemaccana commented:
We could just add a polyfill for legacy browsers. This keeps the code clean and can be deleted quickly when you finally drop IE. :^)

developit commented:
I'm too obsessed with file size to go that route 😜

mikemaccana commented:
Developit: it's literally only a few characters 

    if (!String.prototype.includes) {
      String.prototype.includes = function(search, start) {
        'use strict';
        if (typeof start !== 'number') {
          start = 0;
        }
        
        if (start + search.length > this.length) {
          return false;
        } else {
          return this.indexOf(search, start) !== -1;
        }
      };
    }

See https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/String/includes . It's also way more readable than ye olde "if index is greater than or equal to zero".

Commit:
Use a better seperator. Handle newlines, various other seperators.

Commit:
Remove AMD and global crap from build

Commit:
Rebuild and bump version

Commit:
Remove maps, HTML, bower, babel files and other unncessary bits (users of module can pick their own build tools)
Make index.js just an npm module. Nobody cares about UMD, AMD, or TC39 modules.

Commit:
Use my editorconfig

Commit:
Remove jshintrc (we use eslint)

Commit:
Add tests for splitting input

Commit:
Neaten selector function
Fix 'base' and just use document

Commit:
Clean less, jslint etc from package.json
Update README.md

Commit:
Run through prettier-eslint - fixes odd conditionals

Commit:
Use array for array.

Commit:
Unminiify variables

Commit:
More cleanup

Commit:
Scope

Commit:
Update readme

Commit:
Add license properly

Commit:
Fix bad import and bump ver
